src/saveToCase.js

```diff
@@ -0,0 +1,475 @@
+define([
+    'vellum/form',
+    'jquery',
+    'underscore',
+    'vellum/mugs',
+    'vellum/parser',
+    'vellum/tree',
+    'vellum/util',
+    'vellum/widgets',
+    'tpl!vellum/templates/widget_update_case',
+    'vellum/core'
+], function (
+    form_,
+    $,
+    _,
+    mugs,
+    parser,
+    Tree,
+    util,
+    widgets,
+    widget_update_case
+){
+    function createsCase(mug) {
+        return mug ? mug.p.use_create : false;
+    }
+
+    function closesCase(mug) {
+        return mug ? mug.p.use_close : false;
+    }
+
+    function updatesCase(mug) {
+        return mug ? mug.p.use_update : false;
+    }
+
+    function addSetValue(mug) {
+        var path = mug.absolutePath;
+
+        if (createsCase(mug)) {
+            mug.form.addSetValue('xforms-ready', path + "/case/@case_id", mug.p.case_id);
+        }
+    }
+
+    var propertyWidget = function (mug, options) {
+            var widget = widgets.normal(mug, options),
+                id = options.id;
+
+            widget.kvInput = $('<div class="control-row" />').attr('name', id);
+
+            widget.getControl = function () {
+                return widget.kvInput;
+            };
```

Comment:
Not sure if you want to future-proof this for the new [mug messages framework](https://github.com/dimagi/Vellum/blob/3b17849889117e21264739ee95779fe8af6c29af/src/widgets.js#L124-L138), or if you want to come back and do that later? I think all you need to do is define `widget.getMessagesContainer` here (and put the messages container div inside the `kvInput` div.